### PR TITLE
[CR] Improve mutation category calculation

### PR DESCRIPTION
**Pull request by RadHazard**
Fix for #21909

The new formula was implemented as described in that issue, and works as follows:  For each mutation the player has, the player gets 8 points in that mutation's categories.  Direct prerequisite and replacement mutations are worth 4 points each.  Each additional step divides the value by 2, until mutations 4+ steps in either direction in the tree count for nothing at all.  The new algorithm builds the dependency tree before calculating the point values, so only the "closest" instance of each mutation is counted.

Preliminary testing looks good.  Strong + Fat Deposits actually gives you the most points in the Ursine category as you'd expect:
New category strengths

It's also much easier to cross the Ursine threshold, bringing it much closer to the other mutation lines.  In fact, it may be a bit *too* easy.  The numbers will likely need tweaking to balance it out.

### CODE REVIEW

**Comment by RadHazard** [comment]
I'm working on adding some basic sanity-check tests around mutations now.  The ones I'm currently working on:

* If you have all pre-threshold mutations in a category, that category is your strongest
* If you have all pre- and post-threshold mutations in a category, that category is your strongest
* If you have all pre-threshold mutations in a category, the (category / total) strength ratio is within a sensible range (0.2 - 0.4?)

I think that should cover the basics, but if there's anything I missed let me know

**Comment by RadHazard** [comment]
Tests have been added.  I had to make one change to get them to pass:
Claws now have MUTCAT_LUPINE.  This buffs wolf characters up above the 20% minimum threshold chance.  This change shouldn't have any other effects, since wolves end up with claws anyway due to it being a requirement of the Paws mutation.

**Comment by Coolthulhu** [comment]
The feature itself looks good, well documented and all, but the test breaks other tests.
It's not the fault of the test itself, but rather other tests assuming that everything is "clean" when the test begins. The affected test is `"visitable_remove"`, which also uses `g->u`.

It would be more robust to use a temporary NPC here instead of `g->u`. Tests that don't use `g->u` probably should stop using it.

File: tests/mutation_test.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#include "catch/catch.hpp"
+
+#include "mutation.h"
+
+#include "game.h"
+#include "player.h"
+
+void clear_mutations( player &p )
```
Comment (Coolthulhu) [inline tests/mutation_test.cpp]: If you used a temporary NPC instead of a complete player, this wouldn't be needed.

**Comment by RadHazard** [comment]
Tests have been updated to use temporary NPCs

**Pull request merged**